dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/JarLinkageReport.java

```diff
@@ -110,35 +108,26 @@ public String formatByGroup() {
             .addAll(groupedFieldErrors.keySet())
             .build();
 
-    String missingMembers = getTotalErrorCount() == 1 ? "1 linkage error in "
-        : getTotalErrorCount() + " linkage errors in ";
-    String inClasses = combinedKeys.size() == 1 ? "1 class\n" : combinedKeys.size() + " classes\n";
-    builder.append(missingMembers + inClasses);
+    ImmutableMultimap.Builder<LinkageErrorCause, String> builder = ImmutableMultimap.builder();
     for (LinkageErrorCause key : combinedKeys) {
-      builder.append(indent);
-      builder.append(key);
       List<StaticLinkageError<? extends SymbolReference>> allErrorsForKey =
           Lists.newArrayList(
               Iterables.concat(
                   groupedClassErrors.get(key),
                   groupedMethodErrors.get(key),
                   groupedFieldErrors.get(key)));
-      String sourceClassJoined =
+      builder.putAll(
+          key,
           allErrorsForKey.stream()
               .map(StaticLinkageError::getReference)
               .map(SymbolReference::getSourceClassName)
-              .distinct()
-              .collect(Collectors.joining(", "));
-
-      builder.append("Referenced by: ");
-      builder.append(sourceClassJoined);
-      builder.append("\n");
+              .map(className -> className.split("\\$")[0]) // Removing duplicate inner classes
+              .collect(toImmutableSet()));
     }
-    return builder.toString();
+    return builder.build();
   }
 
   public int getTotalErrorCount() {
-    return getMissingClassErrors().size() + getMissingMethodErrors().size()
-        + getMissingFieldErrors().size();
+    return getCauseToSourceClasses().size();
```

Comment:
This avoids counting duplicate inner classes.

Reply:
The name seems not accurate then. This is no longer returning the number of errors.